runtime/compiler/z/codegen/ReduceSynchronizedFieldLoad.cpp

```diff
@@ -284,7 +284,9 @@ ReduceSynchronizedFieldLoad::performOnTreeTops(TR::TreeTop* startTreeTop, TR::Tr
          TR::Node* monentNode = iter.currentNode()->getOpCodeValue() == TR::monent ?
             iter.currentNode() :
             iter.currentNode()->getFirstChild();
-
+         // If value types are enabled and the monitor is confirmed non value type, than only do this optimization.
```

Comment:
@fjeremic  We will not carry out further test in the loop if we find valueType is enabled and node can be of value type.

Reply:
The comment is not very useful as the `if` statement pretty much says the same thing. Also the `!` should be propagated inside the parenthesis. How about:

```
// Locking on value types is prohibited by the spec., so this optimization can only be performed if we are certain (at compile time) the locking object is not a value type
if (TR::Compiler->om.areValueTypesEnabled() && cg->isMonitorValueType(node) != TR_no)
   continue;
```

Reply:
Fixed in https://github.com/eclipse/openj9/compare/247123bcd0bf7c87262e954972690bf87f08e56a..194c9838319153c006f53f64cfd8db44d2a78ad5